core/vocabs/loader/loader.factor

```diff
@@ -56,7 +56,8 @@ PRIVATE>
     ] cache ;
 
 : vocab-exists? ( name -- ? )
-    dup lookup-vocab [ ] [ find-vocab-root ] ?if ;
+    [ dup lookup-vocab [ ] [ find-vocab-root ] ?if ]
+    [ dup bad-vocab-name? [ 2drop f ] [ rethrow ] if ] recover ;
```

Comment:
Actually, ` lookup-vocab` also calls `vocab-name`, so they both throw here.

If you see usage of `vocab-exists?`, there are only two places. Both can use the fix, and it seems consistent with the word name: an invalid vocab name is a just a special case of a non-existing vocab, so it should just return `f`. We are not obliged to throw reasons _why_ the vocab doesn't exist.